src/http/request.hpp

```diff
@@ -98,7 +98,7 @@ class Request : public NonCopyable, public NonMovable {
             // tor_socks_port takes precedence because it's more specific
             if (settings.find("tor_socks_port") != settings.end()) {
                 std::string proxy = "127.0.0.1:";
-                proxy += settings["tor_socks_port"];
+                proxy += settings["tor_socks_port"].as<std::string>();
```

Comment:
Also here I think we can rely on type deduction and allow to assign `Item` to `std::string`.